File: python_pip_examples/ticker.py
from bitcoinaverage import RestfulClient


if __name__ == '__main__':
    secret_key = '' or input('Enter your secret key: ')
    public_key = '' or input('Enter your public key: ')

    restful_client = RestfulClient(secret_key, public_key)

    ticker_all_local = restful_client.ticker_all_local(crypto='LTC', fiat='GBP')
    print('ticker all local')
    print(ticker_all_local)

    ticker_all_global = restful_client.ticker_all_global(crypto='BTC', fiat='USD,EUR')
    print('ticker all global')
    print(ticker_all_global)

    ticker_local_per_symbol = restful_client.ticker_local_per_symbol('BTCUSD')
    print('Local Ticker for BTCUSD')
    print(ticker_local_per_symbol)

    ticker_global_per_symbol = restful_client.ticker_global_per_symbol('BTCUSD')
    print('Global Ticker for BTCUSD')
    print(ticker_global_per_symbol)

    ticker_short_local = restful_client.ticker_short_local()
    print('Local Ticker Short')
    print(ticker_short_local)

    ticker_short_global = restful_client.ticker_short_global()
    print('Global Ticker Short')
    print(ticker_short_global)

    print('Generate price index for BTCUSD based only on the data from Bitstamp and Kraken')
    ticker_custom_include = restful_client.ticker_custom_include('BTCUSD', 'bitstamp,kraken')
    print(ticker_custom_include)

    print('Generate price index for BTCUSD based on the data from all exchanges except Bitstamp and Kraken')
    ticker_custom_exclude = restful_client.ticker_custom_exclude('BTCUSD', 'bitstamp,kraken')
    print(ticker_custom_exclude)

    ticker_changes_all_local = restful_client.ticker_changes_all_local()
    print('Local Ticker values with changes')
    print(ticker_changes_all_local)

    ticker_changes_local_btcusd = restful_client.ticker_changes_local('BTCUSD')
    print('Ticker including changes (for BTCUSD):')
    print(ticker_changes_local_btcusd)

    ticker_changes_global_btceur = restful_client.ticker_changes_global('BTCEUR')
    print('Ticker including changes (for BTCEUR):')
    print(ticker_changes_global_btceur)

File: node_npm_examples/ticker.js
const ba = require('bitcoinaverage');

var publicKey = 'yourPublicKey';
var secretKey = 'yourSecretKey';
var restClient = ba.restfulClient(publicKey, secretKey);

// Get local ticker for BTCUSD and ETHUSD
restClient.tickerAllLocal('BTC,ETH', 'USD', function (response) {
    console.log(response);
});

// Get global ticker for BTCUSD and BTCEUR
restClient.tickerAllGlobal('BTC', 'USD,EUR', function(response) {
    console.log(response);
});

// Get local ticker for BTCUSD
restClient.tickerLocalPerSymbol('BTCUSD', function (response) {
    console.log(response);
});

// Get global ticker for BTCGBP
restClient.tickerGlobalPerSymbol('BTCGBP', function (response) {
    console.log(response);
});


// 4 examples for short ticker using the local market:
// 1. Get local short ticker for BTCUSD and BTCEUR
restClient.tickerShortLocal('BTC', 'USD,EUR', function (response) {
    console.log(response);
});

// 2. Get local short ticker for BTC and all supported fiat currencies
restClient.tickerShortLocal('BTC', '', function (response) {
    console.log(response);
});

// 3. Get local short ticker for all crypto currencies, represented in USD
restClient.tickerShortLocal('', 'USD', function (response) {
    console.log(response);
});

// 4. Get local short ticker for all available crypto-fiat pairs
restClient.tickerShortLocal('', '', function (response) {
    console.log(response);
});


// 4 examples for short ticker using the global market:
// 1. Get global short ticker for BTCUSD and BTCEUR
restClient.tickerShortGlobal('BTC', 'USD,EUR', function (response) {
    console.log(response);
});

// 2. Get global short ticker for BTC and all supported fiat currencies
restClient.tickerShortGlobal('BTC', '', function (response) {
    console.log(response);
});

// 3. Get global short ticker for all crypto currencies, represented in USD
restClient.tickerShortGlobal('', 'USD', function (response) {
    console.log(response);
});

// 4. Get global short ticker for all available crypto-fiat pairs
restClient.tickerShortGlobal('', '', function (response) {
    console.log(response);
});

// Custom ticker
// Generate price index for BTCUSD based only on the data from Bitstamp and Kraken
restClient.tickerCustomInclude('BTCUSD', 'bitstamp,kraken', function (response) {
    console.log(response);
});

// Generate price index for BTCUSD based on the data from all exchanges except Bitstamp and Kraken
restClient.tickerCustomExclude('BTCUSD', 'bitstamp,kraken', function (response) {
    console.log(response);
});

// Get ticker and changes for all supported symbols
restClient.tickerChangesAllLocal(function(response) {
    console.log(response);
});

// Get Local Ticker and changes only for BTCUSD
restClient.tickerChangesLocal('BTCUSD', function (response) {
    console.log(response);
});

// Get Global Ticker and changes only for ETHUSD
restClient.tickerChangesGlobal('ETHUSD', function (response) {
    console.log(response);
});